js/stateless.js/src/instructions/compressed-pda.ts

```diff
@@ -0,0 +1,138 @@
+import { Program, AnchorProvider, setProvider } from "@coral-xyz/anchor";
+import {
+  PublicKey,
+  TransactionInstruction,
+  Keypair,
+  Connection,
+} from "@solana/web3.js";
+import { IDL, PspCompressedPda } from "../idls/psp_compressed_pda";
+import { confirmConfig } from "../constants";
+import { useWallet } from "../wallet";
+import { UtxoWithMerkleProof } from "../state";
+
+export type CompressedTransferParams = {
+  /** Utxos with lamports to spend as transaction inputs */
+  fromBalance: UtxoWithMerkleProof[];
+  /** Solana Account that will receive transferred compressed lamports as utxo  */
+  toPubkey: PublicKey;
+  /** Amount of compressed lamports to transfer */
+  lamports: number | bigint;
+};
+
+/**
+ * Create compressed account system transaction params
+ */
+export type CreateCompressedAccountParams = {
+  /**
+   * Optional utxos with lamports to spend as transaction inputs.
+   * Not required unless 'lamports' are specified, as Light doesn't
+   * enforce rent on the protocol level.
+   * */
+  fromBalance: UtxoWithMerkleProof[];
+  /** Public key of the created account */
+  newAccountPubkey: PublicKey;
+  /** Amount of lamports to transfer to the created compressed account */
+  lamports: number | bigint;
+  /** Public key of the program to assign as the owner of the created compressed account */
+  programId: PublicKey;
+};
+
+/**
+ * Example usage:
+ * ```typescript
+ * const tx = await LightSystemProgram.transfer({
+ *  fromBalance: [
+ *   {
+ *      owner: new PublicKey("..."),
+ *      data: [],
+ *      lamports: 2000000000n,
+ *      leafIndex: 0n,
+ *      hash: 0n,
+ *      merkletreeId: 0n,
+ *      merkleProof: [],
+ *   },
+ * ],
+ * toPubkey: new PublicKey("..."),
+ * lamports: 1000000000n,
+ * });
+ * ```
+ */
+export class LightSystemProgram {
+  /**
+   * @internal
+   */
+  constructor() {}
+
+  /**
+   * Public key that identifies the CompressedPda program
+   */
+  static programId: PublicKey = new PublicKey(
+    // TODO: replace with actual program id
+    // can add check to ensure its consistent with the idl
+    "11111111111111111111111111111111"
+  );
+
+  private static _program: Program<PspCompressedPda> | null = null;
+
+  /**
+   * Accessor for the program. Ensures the program is initialized before returning.
+   */
+  static get program(): Program<PspCompressedPda> {
```

Comment:
technically either achieve the same thing but I favor using a getter since in this context we're accessing a shared `_program` property of a class singleton (instead of a class instance), so using a getter is semantically more appropriate, esp. since there's no need for parameters, making it clear that we're rather just accessing a pre-existing, possibly lazily-inited property with `PspCompressedPda.program` 

 for ref. this is akin to how Keypair is implemented in web3js:

```
 /**
   * The public key for this keypair
   *
   * @returns {PublicKey} PublicKey
   */
  get publicKey(): PublicKey {
    return new PublicKey(this._keypair.publicKey);
  }
```